### DISCUSSION TITLE
Node Hyperlinks: Traverse a graph

### DISCUSSION THREAD

**Discussion by joelonsql** [terrastruct/d2#486] [Feature Requests]
I'm considering D2 to auto-generate diagrams of PostgreSQL database schemas.

When you are exploring a new database, you might start the journey from some given object, such as a table, and then look at all connected objects, such as other tables with references to the given table, and tables that the table references.

To allow the graph to be recursively traversed, I would like each node to be clickable.

This could be achieved by providing a way to specify a hyperlink, similar to <a href="..."/></a> in HTML.
In general, that would allow any node to function as a link, effectively making the user's browser leave the graph and visit the URL.
In my case the new web page loaded would be a new graph, where the clicked node would then be in center, showing all directly connected objects.

**Comment by alixander** [comment]
@joelonsql this person made a basic version of this: https://github.com/zekenie/d2-erd-from-postgres . If you'd like to make a more robust/maintained version, let me know how I can help.

Re: clickable -- we will support this in the next release! https://github.com/terrastruct/d2/issues/417
